### PR TITLE
test  for ImageValidator 

### LINKED ISSUE #9: Create unit test to ImageValidator.java
### Create Unit Tests for `ImageValidator` (`GreenCityMVP`)

**File:**  
`core/src/main/java/greencity/validator/ImageValidator.java`

### Description

Develop a set of unit tests for the `ImageValidator` class, which validates uploaded image files.  
This validator is used with the custom annotation `@ImageValidation` to ensure that uploaded files are of allowed image MIME types.

### Input Data

- Valid types: `"image/jpeg"`, `"image/png"`, `"image/jpg"`
- Input: `MultipartFile` instance (can be `null`, valid, or invalid)

### Test Scenarios

- [x] `isValid(null)` → should return `true` (image is optional)
- [x] `isValid(...)` with `image/png` → should return `true`
- [x] `isValid(...)` with `image/jpeg` → should return `true`
- [x] `isValid(...)` with `image/jpg` → should return `true`
- [x] `isValid(...)` with `image/gif`, `application/pdf`, etc. → should return `false`
- [x] Boundary test: file with `null` content type → should return `false`

###  Acceptance Criteria

- [x] All tests pass successfully (`JUnit 5`)
- [x] `MultipartFile` is mocked using `Mockito`
- [x] Tests are clear and follow Arrange-Act-Assert (AAA) pattern
- [x] Method `isValid(...)` is fully covered:
  - True for valid image types
  - False for disallowed types
  - True for null file input
- [ ] **100% test coverage** for `ImageValidator`

### PR DESCRIPTION
**Pull request by oleksandrYeriemieiev**
Covered all test scenarios: null input, valid MIME types, invalid MIME types, null content type.
Used Mockito for `MultipartFile` mock.
100% coverage for `isValid(...)` method.

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive tests for image validation, covering various file types and scenarios to ensure correct validation behavior.

### CODE REVIEW

**Review by MykytaSirobaba — CHANGES_REQUESTED** [review]
Hi! The test is valid, but you could improve it by using **ParameterizedTest** and **ValueSource**. This allows you to pass input data directly via the annotation, for example: **ValueSource(strings = {"image/png", "image/jpeg", "image/jpg"})**.
This way, you can reduce redundancy by merging similar test cases that return true into a single test method, and do the same for those that return false.

**Comment by oleksandrYeriemieiev** [comment]
Updated according to review and nitpick:
Applied parameterized tests for valid and invalid MIME types.
Added edge case test.

Please check!

**Review by MykytaSirobaba — CHANGES_REQUESTED** [review]
Hi! Great job on this. However, it looks like the test coverage isn't 100% yet. Could you please take a look and address that? Thanks!

Знімок екрана 2025-07-01 о 11 57 16

File: core/src/test/java/greencity/validator/ImageValidatorTest.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"", " ", "image/", "IMAGE/PNG", "image/jpeg; charset=utf-8"})
+    void testIsValid_EdgeCases_ReturnsFalse(String contentType) {
+        MultipartFile file = Mockito.mock(MultipartFile.class);
+        Mockito.when(file.getContentType()).thenReturn(contentType);
+
+        assertFalse(validator.isValid(file, null));
+    }
```
Comment (Kostya-Pav) [inline core/src/test/java/greencity/validator/ImageValidatorTest.java:51]: You should combine these two tests since they check the same behavior.

Comment (oleksandrYeriemieiev) [reply core/src/test/java/greencity/validator/ImageValidatorTest.java:51]: Kept invalid mime types and edge cases in separate tests on purpose.
They check different groups: standard invalid types (image/gif, null) and edge format (empty, whitespace, uppercase..)
If one fails, it’s clear which  caused it — helps to debug .
Hope that explains the idea!

**Comment by oleksandrYeriemieiev** [comment]
@MykytaSirobaba Hi! 
Test covers the isValid() method fully.
The only reason coverage is not 100% is the initialize() method in ImageValidator — it’s empty by design and has no logic to test.
So nothing is missing for this validator.
Thanks for the feedback!

**Review by natri0 — APPROVED** [review]

**Review by Linney34 — APPROVED** [review]

**Review by AndriiSynytsia — APPROVED** [review]

**Review by Denys8807 — APPROVED** [review]

**Review by MykytaSirobaba — APPROVED** [review]
